spec/controllers/user_reports_controller_spec.rb

```diff
@@ -0,0 +1,158 @@
+require "rails_helper"
+
+RSpec.describe UserReportsController, type: :controller do
+  let(:user){FactoryBot.create :user}
+  let(:user2){FactoryBot.create :user, full_name: "Name 2", email: "test2@mail.com"}
+  let(:user3){FactoryBot.create :user, full_name: "Name 3", email: "test3@mail.com"}
+  let(:course){FactoryBot.create :course}
+  let!(:user_course){FactoryBot.create :user_course, course: course, user: user}
+  let!(:user_course2){FactoryBot.create :user_course,
+                        course: course, user: user2}
+  let(:report_1){FactoryBot.create :user_report, content: "The first report",
+                   date: 2.days.ago, course: course, user: user}
+  let(:report_2){FactoryBot.create :user_report, content: "The first report",
+                   date: 2.days.ago, course: course, user: user2}
+
+  before{login user}
+
+  describe "GET /index" do
+    context "when user has courses" do
+      before{get :index}
+
+      it {expect(response).to render_template :index}
+    end
+
+    context "when user don't have course" do
+      before do
+        login user3
+        get :index
+      end
+
+      it {expect(response).to redirect_to root_url}
+    end
+  end
+
+  describe "GET /new" do
+    before{get :new}
+
+    it {expect(response).to render_template :new}
+    it {expect(assigns(:user_report)).to be_a_new UserReport}
+  end
+
+  describe "POST /FactoryBot.create" do
+    context "when user report informatuon is valid" do
+      before do
+        post :create, params: {user_report: {course_id: course.id,
+          date: Time.now, content: "Content"}}
+      end
+
+      it {expect(flash[:success]).to eq I18n.t("success")}
+      it {expect(response).to redirect_to user_reports_path}
+    end
+
+    context "when user report informatuon is invalid" do
+      before do
+        post :create, params: {user_report: {course_id: course.id,
+          date: Time.now, content: ""}}
+      end
+
+      it {expect(response).to render_template :new}
+    end
+  end
+
+  describe "GET /show" do
+    context "when report not exist" do
+      before{get :show, params: {id: -1}}
+
+      it {expect(flash[:danger]).to eq I18n.t("report_not_found")}
+      it {expect(response).to redirect_to user_reports_path}
+    end
+
+    context "when report existed but not owned by current_user" do
+      before{get :show, params: {id: report_2.id}}
+
+      it {expect(flash[:danger]).to eq I18n.t("report_invalid")}
```

Comment:
có thể là thế
hoặc to be present cũng dc em nha :p

Reply:
Vâng ạ.